File: modules/rest/src/main/java/dev/httpmarco/polocloud/modules/rest/configuration/UsersConfiguration.java
package dev.httpmarco.polocloud.modules.rest.configuration;

import dev.httpmarco.polocloud.modules.rest.auth.user.User;
import dev.httpmarco.polocloud.node.util.StringUtils;
import lombok.Getter;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@Getter
@Accessors(fluent = true)
public class UsersConfiguration {

    private final String secret;
    private final List<User> users;

    public UsersConfiguration() {
        this.secret = StringUtils.randomString(170);
        this.users = new ArrayList<>();
    }
}

File: modules/rest/src/main/kotlin/dev/httpmarco/polocloud/modules/rest/config/UsersConfiguration.kt
package dev.httpmarco.polocloud.modules.rest.config

import dev.httpmarco.polocloud.modules.rest.auth.role.Role
import dev.httpmarco.polocloud.modules.rest.auth.user.User
import dev.httpmarco.polocloud.modules.rest.utils.generateRandom

data class Users(
    val secret: String = generateRandom(),
    val roles: MutableList<Role> = mutableListOf(),
    val users: MutableList<User> = mutableListOf()
) : Config